src/core/internal/cppdemangle.d

```diff
@@ -0,0 +1,311 @@
+/**
+* This module provides functions to demangle C++ symbols
+*
+* For POSIX platforms the function `__cxa_demangle` present in Itanium C++ ABI does the demangling, therefore this module
+* depends on the C++ standard library which must be linked dynamically or available for research through the dynamic loader.
+*
+* For Windows platform the function `UnDecorateSymbolName` present in the Debug Help Library does the demangling,
+* which should be available for all versions of Windows supported by D.
+*
+* Copyright: Copyright © 2020, The D Language Foundation
+* License: $(HTTP www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+* Authors: Ernesto Castellotti
+* Source: $(DRUNTIMESRC core/internal/_cppdemangle.d)
+* See_Also:
+*      https://itanium-cxx-abi.github.io/cxx-abi/abi.html#demangler
+*      https://gcc.gnu.org/onlinedocs/libstdc++/libstdc++-html-USERS-4.3/a01696.html
+*      https://libcxxabi.llvm.org/spec.html
+*      https://docs.microsoft.com/en-us/windows/win32/api/dbghelp/nf-dbghelp-undecoratesymbolname
+*/
+module core.internal.cppdemangle;
+
+/**
+* Demangles C++ mangled names.
+*
+* The output buffer and return will be contains the demangled name of C++ symbol if the demangling is successful.
+*
+* This function will fail and the mangled name in input will be returned if:
```

Comment:
I think the "return input on failure" part should be mentioned in the previous sentence as it describes the functions output.